routing/road_graph.cpp

```diff
@@ -59,39 +58,35 @@ string DebugPrint(Junction const & r)
 
 // Edge ------------------------------------------------------------------------
 // static
-Edge Edge::MakeFake(Junction const & startJunction, Junction const & endJunction, bool partOfReal)
+Edge Edge::MakeFake(Junction const & startJunction, Junction const & endJunction)
 {
-  return MakeFakeWithForward(startJunction, endJunction, partOfReal, true /* forward */);
+  Edge e{FeatureID(), true /* forward */, 0 /* segId */, startJunction, endJunction};
+  e.m_type = Type::FakeWithNoRealPart;
+  return e;
 }
 
-// static
-Edge Edge::MakeFakeForTesting(Junction const & startJunction, Junction const & endJunction,
-                              bool partOfReal, bool forward)
-{
-  return MakeFakeWithForward(startJunction, endJunction, partOfReal, forward);
-}
-
-// static
-Edge Edge::MakeFakeWithForward(Junction const & startJunction, Junction const & endJunction,
-                               bool partOfReal, bool forward)
+Edge Edge::MakeFake(Junction const & startJunction, Junction const & endJunction,
+                    Edge const & prototype)
 {
-  Edge edge(FeatureID(), forward, 0 /* segId */, startJunction, endJunction);
-  edge.m_partOfReal = partOfReal;
-  return edge;
+  auto e = prototype;
+  e.m_startJunction = startJunction;
+  e.m_endJunction = endJunction;
+  e.m_type = e.HasRealPart() ? Type::FakeWithRealPart : Type::FakeWithNoRealPart;
+  return e;
 }
 
-Edge::Edge() : m_forward(true), m_segId(0) { m_partOfReal = !IsFake(); }
+Edge::Edge() : m_type(Type::FakeWithNoRealPart), m_forward(true), m_segId(0) {}
```

Comment:
Could you please add default initializers and then, declare default ctor with default keyword?

Reply:
Done.